content/en/docs/File Format/implementationstatus.md

```diff
@@ -0,0 +1,178 @@
+| Modular encryption                        |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Page CRC32 checksum                       |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Modular encryption                        |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+
+High level data API-s for parquet feature usage
+===============================================
+
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Format                                       | C++   | Python | Java   | Go    | Rust  |
+|                                              |       |        |        |       |       |
++==============================================+=======+========+========+=======+=======+
+| Hive-style partitioning                      |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Partition pruning on the partition column    |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| External column data                         |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| RowGroup Sorting column                      |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Read / Write RowGroup metadata and data (1)  |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| RowGroup pruning using statistics            |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Read / Write page metadata and data (2)      |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Page pruning using projection pushdown       |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Page pruning using statistics                |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Page pruning using bloom filter              |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+
+* \(1) Ability to construct RowGroup objects from existing RowGroups or raw values (eg. without decoding or decompressing lower level data when read)
```

Comment:
I'm honestly not sure what this means. Reassemble a new Parquet file from existing undecoded column chunks?

Reply:
Yes, that was my intention. AFAIK Both rust and Java solutions support it to some degree

Reply:
Removed for now